runtime/list.go

```diff
@@ -291,6 +300,29 @@ func listNE(f *Frame, v, w *Object) (*Object, *BaseException) {
 	return listCompare(f, toListUnsafe(v), w, NE)
 }
 
+func listPop(f *Frame, args Args, kwargs KWArgs) (*Object, *BaseException) {
+	argc := len(args)
+	expectedTypes := []*Type{ListType, IntType}
+	if argc == 1 {
+		expectedTypes = expectedTypes[:1]
+	}
+	if raised := checkMethodArgs(f, "pop", args, expectedTypes...); raised != nil {
+		return nil, raised
+	}
+	i := -1
+	if argc == 2 {
+		i = toIntUnsafe(args[1]).Value()
```

Comment:
It's actually a bit more complicated than that :\

```
>>> class A(object):
...  def __int__(self):
...   return 1
...
>>> [1,2,3].pop(A())
2
```

I think the easiest thing to do is something like:

```
o, raised := IntType.Call(f, Args{args[1]}, nil)
if raised != nil {
  return nil, raised
}
i = toIntUnsafe(o).Value()
```

That will handle int and long properly as well as types that define `__int__`.